Implemented PlayerRanking.java, Fixed Bugs in UserProfile and Added test cases

tomwurdfz opened the pull request:
Implemented methods getRankByGames(), getRankByWins() and getRankByWinrate in PlayerRanking.java

tomwurdfz committed:
Implemented methods getRankByGames(), getRankByWins() and getRankByWinrate in PlayerRanking.java

SarahX43 reviewed (dismissed):
Thanks for implementing the classes. I will create a UI page that presents the player rankings accordingly. It would be great if we can have more documentation so the codes can be more readable.

tomwurdfz committed:
Implemented methods getRankByGames(), getRankByWins() and getRankByWinrate in PlayerRanking.java, fixed semicolumns

tomwurdfz committed:
Fixed the constructor in UserProfile

tomwurdfz committed:
Updated UserProfile.java and fixed the divided by zero bug when calculating the winrate, added test cases for all methods in UserProfile.java.
Fixed all warnings.

tomwurdfz commented:
Updated UserProfile.java and fixed the divided by zero bug when calculating the winrate, added test cases for all methods in UserProfile.java.
Fixed all warnings in those two classes.

SarahX43 reviewed (approved):
the user profile use case is ready, we just need to connect it with presenter and controller.
